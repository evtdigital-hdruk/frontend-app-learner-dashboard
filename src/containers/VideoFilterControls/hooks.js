import React from 'react';
import { useToggle } from '@openedx/paragon';

import { StrictDict } from 'utils';
import track from 'tracking';
import { reduxHooks } from 'hooks';

import * as module from './hooks';

export const state = StrictDict({
  target: (val) => React.useState(val), // eslint-disable-line
});

/**
 * Sets up a toggle for the modal as well as helper functions for handling changes to the form controls.
 *
 * @param {array} filters Currently active video filters
 * @param {function} setSortBy Set function for sorting the video list
 * @returns {object} data and functions for managing the VideoFilterControls component
 */
export const useVideoFilterControlsData = ({
  filters,
  setSortBy,
}) => {
  const [isOpen, toggleOpen, toggleClose] = useToggle(false);
  const [target, setTarget] = module.state.target(null);

  const addFilter = reduxHooks.useAddVideoFilter();
  const removeFilter = reduxHooks.useRemoveVideoFilter();

  const handleFilterChange = ({ target: { checked, value } }) => {
    const update = checked ? addFilter : removeFilter;
    update(value);
  };
  const handleSortChange = ({ target: { value } }) => {
    setSortBy(value);
  };

  const open = () => {
    track.filter.filterClicked();
    toggleOpen();
  };

  const close = () => {
    track.filter.filterOptionSelected(filters);
    toggleClose();
  };

  return {
    isOpen,
    open,
    close,
    target,
    setTarget,
    handleFilterChange,
    handleSortChange,
  };
};

export default useVideoFilterControlsData;
