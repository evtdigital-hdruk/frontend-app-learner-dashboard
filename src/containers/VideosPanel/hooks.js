import React from 'react';

import queryString from 'query-string';

import { ListPageSize, SortKeys } from 'data/constants/app';
import { reduxHooks } from 'hooks';
import { StrictDict } from 'utils';

import * as module from './hooks';

export const state = StrictDict({
  sortBy: (val) => React.useState(val), // eslint-disable-line
});

/**
 * Filters are fetched from the store and used to generate a list of "visible" courses.
 * Other values returned and used for the layout of the CoursesPanel component are:
 * the current page number, the sorting method, and whether or not to enable filters and pagination.
 *
 * @returns data for the CoursesPanel component
 */
export const useVideoListData = () => {
  const videoFilters = reduxHooks.useVideoFilters();
  const removeVideoFilter = reduxHooks.useRemoveVideoFilter();
  const videoPageNumber = reduxHooks.useVideoPageNumber();
  const setVideoPageNumber = reduxHooks.useSetVideoPageNumber();

  const [sortBy, setSortBy] = module.state.sortBy(SortKeys.enrolled);

  const querySearch = queryString.parse(window.location.search, { parseNumbers: true });

  const { numPages, visibleVideoList } = reduxHooks.useCurrentVideoList({
    sortBy,
    videoFilters,
    pageSize: querySearch?.disable_pagination === 1 ? 0 : ListPageSize,
  });

  const handleRemoveFilter = (filter) => () => removeVideoFilter(filter);

  return {
    videoPageNumber,
    numPages,
    setVideoPageNumber,
    visibleVideoList,
    filterOptions: {
      sortBy,
      setSortBy,
      videoFilters,
      handleRemoveFilter,
    },
    showFilters: videoFilters.length > 0,
  };
};

export default useVideoListData;
