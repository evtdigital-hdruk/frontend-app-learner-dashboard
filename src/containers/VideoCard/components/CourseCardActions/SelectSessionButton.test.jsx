import { shallow } from 'enzyme';

import { reduxHooks } from 'hooks';
import { htmlProps } from 'data/constants/htmlKeys';
import useActionDisabledState from '../hooks';

import SelectSessionButton from './SelectSessionButton';

jest.mock('hooks', () => ({
  reduxHooks: {
    useUpdateSelectSessionModalCallback: () => jest.fn().mockName('mockOpenSessionModal'),
  },
}));
jest.mock('../hooks', () => jest.fn(() => ({ disableSelectSession: false })));
jest.mock('./ActionButton', () => 'ActionButton');

let wrapper;

describe('SelectSessionButton', () => {
  const props = { cardId: 'cardId' };
  it('default render', () => {
    expect(wrapper).toMatchSnapshot();
    wrapper = shallow(<SelectSessionButton {...props} />);
    expect(wrapper.prop(htmlProps.disabled)).toEqual(false);
    expect(wrapper.prop(htmlProps.onClick).getMockName()).toEqual(
      reduxHooks.useUpdateSelectSessionModalCallback().getMockName(),
    );
  });
  test('disabled states', () => {
    useActionDisabledState.mockReturnValueOnce({ disableSelectSession: true });
    expect(wrapper).toMatchSnapshot();
    wrapper = shallow(<SelectSessionButton {...props} />);
    expect(wrapper.prop(htmlProps.disabled)).toEqual(true);
  });
});
