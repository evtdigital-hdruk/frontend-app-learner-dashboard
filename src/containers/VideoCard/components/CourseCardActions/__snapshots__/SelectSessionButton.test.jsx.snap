// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectSessionButton default render 1`] = `undefined`;

exports[`SelectSessionButton disabled states 1`] = `
<ActionButton
  disabled={false}
  onClick={[MockFunction mockOpenSessionModal]}
>
  Select Session
</ActionButton>
`;
